src/scalars/mod.rs

```diff
@@ -0,0 +1,20 @@
+use async_graphql_value::ConstValue;
+
+pub use crate::scalars::email::Email;
+
+mod email;
+
+pub const CUSTOM_SCALARS: &[&str] = &["Email"];
+
+#[derive(schemars::JsonSchema)]
+/// A wrapper to store all custom scalar types
+pub struct Scalars {
+    pub email: Email,
+}
+
+pub fn get_scalar(name: &str) -> fn(&ConstValue) -> bool {
+    match name {
+        "Email" | "email" => Email::validate,
```

Comment:
We can create a trait `Scalar` with a `validate` method on it, and implement it for `Email` type. We could also then update Scalars to hold a `HasMap<String, dyn Scalar>`.

```rust
trait Scalar {
  fn validate(c: &ConstValue) -> bool;
}
```

Reply:
this function is called while compiling schema. I don't think we need map here